src/instructions/not.rs

```diff
@@ -0,0 +1,63 @@
+use crate::registers::{Register, Registers};
+
+#[derive(Debug)]
+pub struct Not {
+    dr: Register,
+    sr: Register,
+}
+
+impl Not {
+    pub fn from_bits(bits: u16) -> Self {
+        let dr = Register::try_from((bits >> 9) & 0b111).unwrap();
+        let sr = Register::try_from((bits >> 6) & 0b111).unwrap();
+
+        Self { dr, sr }
+    }
+
+    pub fn execute(&self, registers: &mut Registers) {
+        registers.set(self.dr, !registers.get(self.sr));
+        registers.update_flags(self.dr);
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_bits_to_not() {
+        // NOT  R3, R1
+        let bits = 0b1001_011_001_1_11111;
+
+        let instruction = Not::from_bits(bits);
+
+        assert_eq!(instruction.dr, Register::R3);
+        assert_eq!(instruction.sr, Register::R1);
+    }
+
+    #[test]
+    fn test_not() {
+        let mut registers = Registers::default();
+        registers.set(Register::R3, 0);
+
+        let instruction = Not {
+            dr: Register::R3,
+            sr: Register::R1,
+        };
+
+        instruction.execute(&mut registers);
+
+        assert_eq!(registers.get(Register::R3), 0xffff);
+
+        registers.set(Register::R4, 0b1010_1010_0101_0101);
```

Comment:
Didn't know you could write the numbers with the underscore, makes it easier to read

Reply:
Yeah it's cool! You can actually do it with all numeric types.